Sources/TuistGraph/Models/InfoPlist.swift

```diff
@@ -62,58 +126,36 @@ extension InfoPlist: ExpressibleByStringLiteral {
     }
 }
 
-// MARK: - InfoPlist.Value - ExpressibleByStringLiteral
+// MARK: - Entitlements
 
-extension InfoPlist.Value: ExpressibleByStringLiteral {
-    public init(stringLiteral value: String) {
-        self = .string(value)
-    }
-}
-
-// MARK: - InfoPlist.Value - ExpressibleByIntegerLiteral
-
-extension InfoPlist.Value: ExpressibleByIntegerLiteral {
-    public init(integerLiteral value: Int) {
-        self = .integer(value)
-    }
-}
-
-// MARK: - InfoPlist.Value - ExpressibleByIntegerLiteral
-
-extension InfoPlist.Value: ExpressibleByFloatLiteral {
-    public init(floatLiteral value: Double) {
-        self = .real(value)
-    }
-}
+public enum Entitlements: PlistTypesProtocol, Equatable, Codable {
+    // Path to a user defined info.plist file (already exists on disk).
+    case file(path: AbsolutePath)
 
-// MARK: - InfoPlist.Value - ExpressibleByBooleanLiteral
+    // Path to a generated info.plist file (may not exist on disk at the time of project generation).
+    // Data of the generated file
+    case generatedFile(path: AbsolutePath, data: Data)
 
-extension InfoPlist.Value: ExpressibleByBooleanLiteral {
-    public init(booleanLiteral value: Bool) {
-        self = .boolean(value)
-    }
-}
+    // User defined dictionary of keys/values for an info.plist file.
+    case dictionary([String: Plist.Value])
 
-// MARK: - InfoPlist.Value - ExpressibleByDictionaryLiteral
+    // MARK: - Public
 
-extension InfoPlist.Value: ExpressibleByDictionaryLiteral {
-    public init(dictionaryLiteral elements: (String, InfoPlist.Value)...) {
-        self = .dictionary(Dictionary(uniqueKeysWithValues: elements))
+    public var path: AbsolutePath? {
+        switch self {
+        case let .file(path), let .generatedFile(path: path, data: _):
+            return path
+        default:
+            return nil
+        }
     }
 }
 
-// MARK: - InfoPlist.Value - ExpressibleByArrayLiteral
+// MARK: - Entitlements - ExpressibleByStringLiteral
 
-extension InfoPlist.Value: ExpressibleByArrayLiteral {
-    public init(arrayLiteral elements: InfoPlist.Value...) {
-        self = .array(elements)
+extension Entitlements: ExpressibleByStringLiteral {
+    public init(stringLiteral value: String) {
+        self = .file(path: try! AbsolutePath(validating: value)) // swiftlint:disable:this force_try
     }
 }
```

Comment:
Note: this isn't needed here as those are the internal models that are derived from the public API and don't have the same strict backwards source compatibility requirement - so can be omitted all together :)

Reply:
So I just remove the extension?